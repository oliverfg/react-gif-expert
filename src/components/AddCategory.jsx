import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target: {value}}) => {
        setInputValue(value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        const newValue = inputValue.trim();
        if (newValue.length <= 1) return;
        onNewCategory(newValue);
        setInputValue('');
    };

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={ onInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}