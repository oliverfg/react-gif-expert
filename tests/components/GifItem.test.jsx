import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Testing GifItem component', () => { 
    const title = 'Title of item';
    const url = 'https://fakeurl/file.jpg';
    test('should match with snapshot', () => { 
        const { container } = render(
            <GifItem title={title} url={url} />
        );
        
        expect(container).toMatchSnapshot();
     })

    test('should show the image with the src and alt', () => { 
        render(
            <GifItem title={title} url={url} />
        );
        // screen.debug();
        // expect(screen.getByRole('img').alt).toBe(title);
        
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should show the title', () => { 
        render(
            <GifItem title={title} url={url} />
        );

        expect(screen.getByText(title)).toBeTruthy();
    })
 })