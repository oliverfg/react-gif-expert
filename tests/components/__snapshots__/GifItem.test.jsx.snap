// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem component should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title of item"
      src="https://fakeurl/file.jpg"
    />
    <p>
      Title of item
    </p>
  </div>
</div>
`;
